src/component/ReservationForm.js

```diff
@@ -0,0 +1,76 @@
+import { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { useNavigate } from 'react-router-dom';
+import { Form, Button } from 'react-bootstrap';
+import { reserveEvent } from '../redux/event/event';
+import '../css/ReservationForm.css';
+
+const ReservationForm = () => {
+  const details = useSelector((state) => state.events.eventDetails);
+  const dispatch = useDispatch();
+  const reserveEventAction = bindActionCreators(reserveEvent, dispatch);
+  const [eventDate, setEventDate] = useState('');
+  const [eventCity, setEventCity] = useState('');
+  const navigate = useNavigate();
+
+  return (
+    <div className="row bg-image-reservation mx-0 d-flex flex-column justify-content-center align-items-center">
+      <div className="col-sm-6">
+        <h2 className="light-font">Reserve an Event</h2>
+        <Form
+          onSubmit={async (e) => {
+            e.preventDefault();
+            const reservationDetails = {
+              event_id: details.id,
+              date: eventDate,
+              city: eventCity,
+            };
+            await reserveEventAction(reservationDetails);
+            navigate('/attending_events');
```

Comment:
- [ ] Again, this onSubmit code could be transported to a separate function outside of the return statement of the component, what do you think?